python/dgl/contrib/graph_partition_book.py

```diff
@@ -0,0 +1,254 @@
+        self._num_partitions = self._part_meta['num_parts']
+        assert self._num_partitions > 0, 'num_partitions cannot be a negative number.'
+        # Get part_files
+        self._part_files = []
+        for part_id in range(self._num_partitions):
+            assert 'part-{}'.format(part_id) in self._part_meta, "part-{} does not exist".format(part_id)
+            part_files = self._part_meta['part-{}'.format(part_id)]
+            self._part_files.append(part_files)
+        # Get part_graphs
+        self._part_graphs = []
+        for part_id in range(self._num_partitions):
+            graph = load_graphs(self._part_files[part_id]['part_graph'])[0][0]
+            self._part_graphs.append(graph)
+        # Read ip list from ip_config_file
+        self._ip_list = []
+        lines = [line.rstrip('\n') for line in open(ip_config_file)]
+        for line in lines:
+            ip, _, _ = line.split(' ')
+            self._ip_list.append(ip)
+
+
+    def num_partitions(self):
+        """Return the number of partitions.
+
+        Returns
+        -------
+        int
+            number of partitions
+        """
+        return self._num_partitions
+        
+
+    def metadata(self):
+        """Return the partition meta data.
+        
+        The meta data includes:
+
+        * The machine ID.
+        * The machine IP address.
+        * Number of nodes and edges of each partition.
+        
+        Examples
+        --------
+        >>> print(g.get_partition_book().metadata())
+        >>> [{'machine_id' : 0, 'ip': '192.168.8.12', 'num_nodes' : 3000, 'num_edges' : 5000},
+        ...  {'machine_id' : 1, 'ip': '192.168.8.13', 'num_nodes' : 2000, 'num_edges' : 4888},
+        ...  ...]
+        
+        Returns
+        -------
+        list[dict[str, any]]
+            Meta data of each partition.
+        """
+        if len(self._meta_data) == 0:
+            for part_id in range(self._num_partitions):
+                part_info = {}
+                part_info['machine_id'] = part_id
+                part_info['ip'] = self._ip_list[part_id]
+                node_feats = load_tensors(self._part_files[part_id]['node_feats'])
+                edge_feats = load_tensors(self._part_files[part_id]['edge_feats'])
+                part_info['num_nodes'] = len(node_feats)
+                part_info['num_edges'] = len(edge_feats)
+                self._meta_data.append(part_info)
+
+        return self._meta_data
+
+
+    def nid2partid(self, nids):
+        """From global node IDs to partition IDs
+
+        Parameters
+        ----------
+        nids : tensor
+            global node IDs
+
+        Returns
+        -------
+        tensor
+            partition IDs
+        """
+        if self._nid2partid is None:
+            assert 'node_map' in self._part_meta, "cannot get the node map."
+            self._nid2partid = np.load(self._part_meta['node_map'])
```

Comment:
we don't load them in advance? at least DistGraph server loads it when the server starts.
I think you should load it and DistGraph server will call it.

Reply:
Here I just want to use a lazy design-pattern, where the init function will not do much work. But it's OK we can put all the initialization code init __init__.